shivyc/il_cmds/math.py

```diff
@@ -118,6 +118,70 @@ class Mult(_AddMult):
     Inst = asm_cmds.Imul
 
 
+class _BitShiftCmd(ILCommand):
+    """Base class for bitwise shift commands."""
+
+    # The ASM instruction to generate for this command. Override this value
+    # in subclasses.
+    Inst = None
+
+    def __init__(self, output, arg1, arg2): # noqa D102
+        self.output = output
+        self.arg1 = arg1
+        self.arg2 = arg2
+
+    def inputs(self): # noqa D102
+        return [self.arg1, self.arg2]
+
+    def outputs(self): # noqa D102
+        return [self.output]
+
+    def abs_spot_pref(self): # noqa D102
```

Comment:
I forgot to mention this in my previous comments: could you add a `clobber` function to this IL command that returns RCX? Since the body of our make_asm() may change the value of RCX without warning, we want to make sure the register allocator knows about this and does not assign any values to RCX that must be preserved.